File: 0541_easy_hot_reverse-string-ii/solution.py
class Solution:
    def reverseStr(self, s: str, k: int) -> str:
        n = len(s)
        result = ""

        for i in range(0, n, 2 * k):
            
            result += s[i:i + k][::-1]
            if n > i + k:
                result += s[i + k:i + 2 * k]
        
        return result
        

        

File: 0541_easy_hot_reverse-string-ii/solution.java
class Solution {
    public String reverseStr(String s, int k) {
        StringBuilder sb = new StringBuilder();

        int n = s.length();

        for(int i = 0; i < n; i += 2 * k) {
            sb.append(reverseStr(s.substring(i, Math.min(i + k, n))));

            if (n > i + k) {
                sb.append(s.substring(i + k, Math.min(i + 2 * k, n)));
            }
        }

        return sb.toString();
    }

    private String reverseStr(String s) {
        return new StringBuilder(s).reverse().toString();
    }
}